ivi-layermanagement-examples/EGLWLMockNavigation/src/MockNavi.cpp

```diff
@@ -78,27 +118,43 @@ void MockNavi::generateCity()
 
     // generate street z direction
     vec4f streetColor(0.0, 0.0, 0.0, 1.0);
-    vec3f streetPosition = vec3f(0.6 * CITY_GRID_SIZE, 0.0, 0.0);
-    vec3f streetSize = vec3f(CITY_GRID_SIZE * 0.6, 0.0, -CITY_GRID_SIZE * 2.0 * m_houseCount);
-    Street* obj = new Street(streetPosition, streetSize, streetColor, pShader);
+    vec3f streetPosition = vec3f(0.6 * CITY_GRID_SIZE, 0.001, 0.0);
+    vec3f streetSize = vec3f(CITY_GRID_SIZE * 0.6, 0.0, -CITY_GRID_SIZE * 2.0 * m_houseCount * 200.0);
+    Street* obj = nullptr;
+        if(streetTextureLoaded){
+            obj = new Street(streetPosition, streetSize, streetColor, pShaderTexture, streetTexture);
+        }else{
+            obj = new Street(streetPosition, streetSize, streetColor, pShader);
+    }
     m_renderList.push_back(obj);
+    m_updateList.push_back(obj);
 
     // generate streets x direction
     for (int z = 1; z < m_houseCount; ++z)
     {
         vec4f streetColor(0.0, 0.0, 0.0, 1.0);
         vec3f streetPosition = vec3f(0.0, 0.0, 0.6 - z * CITY_GRID_SIZE);
         vec3f streetSize = vec3f(CITY_GRID_SIZE * 3, 0.0, CITY_GRID_SIZE * 0.6);
-        Street* obj = new Street(streetPosition, streetSize, streetColor, pShader);
+        Street* obj = nullptr;
+        if(streetTextureLoaded){
+            obj = new Street(streetPosition, streetSize, streetColor, pShaderTexture, streetTexture);
+        }else{
+            obj = new Street(streetPosition, streetSize, streetColor, pShader);
+        }
         m_renderList.push_back(obj);
         m_updateList.push_back(obj);
     }
 
     // generate car
-    vec3f carPosition(1.4 * CITY_GRID_SIZE, 0.001, -0.3);
+    vec3f carPosition(1.4 * CITY_GRID_SIZE, 0.002, -0.3);
```

Comment:
The position of the car looks like not center. 1.38 of the X coordinate is better than 1.4. I'm not sure how to calculate it properly.

Reply:
The width of the base plate is 3 and the car is in the middle, so if its x position is 1.4 and x size 0.2, it's middle position should be 1.5 and in the middle of the screen. I think, the problem is that the shape of the used texture is not exactly square. 1.39 for the car position with using the texture looks best on my screen :). Now I'm using this position only when a texture could be loaded. And I've fixed car constructor for textures so it now uses the size parameters, which are passed to it.